project/src/test/java/com/google/sps/OpportunitySignupFormHandlerTest.java

```diff
@@ -76,20 +80,42 @@ public void testAddOpportunitySignup_LoggedInOpportunityIdNotSpecified() throws
         .setEnvEmail("test@gmail.com")
         .setEnvAuthDomain("gmail.com");
     Mockito.when(request.getParameter(PARAMETER_OPPORTUNITY_ID)).thenReturn(null);
+    Mockito.when(request.getParameter(PARAMETER_EVENT_ID))
+        .thenReturn(opportunity.getEventId());
 
     opportunitySignupServlet.doPost(request, response);
 
     Mockito.verify(response)
         .sendError(HttpServletResponse.SC_BAD_REQUEST, "Opportunity ID not specified.");
   }
 
+  @Test
+  public void testAddOpportunitySignup_LoggedInEventIdNotSpecified() throws IOException {
+    VolunteeringOpportunity opportunity = TestUtils.newVolunteeringOpportunity();
+    SpannerTasks.insertVolunteeringOpportunity(opportunity);
+    authenticationHelper
```

Comment:
Write a setAuthenticationHelper() kind of helper function for this to abstract away authentication details